azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -741,6 +740,13 @@ func resourceLinuxVirtualMachineUpdate(d *schema.ResourceData, meta interface{})
 		update.Identity = identity
 	}
 
+	if d.HasChange("dedicated_host_id") {
+		shouldUpdate = true
```

Comment:
presumably this also needs to be shut-down and de-allocated - or does the API support live-migration?

Reply:
Yeah, it needs to be deallocated. But shutdown might not be required, the API does not return any error on that and the update was successful